sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -934,7 +952,8 @@ object Limit {
  * See [[Limit]] for more information.
  */
 case class GlobalLimit(limitExpr: Expression, child: LogicalPlan) extends OrderPreservingUnaryNode {
-  override def output: Seq[Attribute] = child.output
+  @transient
+  override lazy val output: Seq[Attribute] = child.output
```

Comment:
Can you keep the old behavior where the call is just pass through?

Reply:
Do you mean all the simple cases like this one?

Reply:
Check the perf difference if we revert these back and only keep the ones that actually generate the values.

Reply:
Yes, I mean simple cases like this. Even if this regresses slightly then I am still in favor of doing it like this. The reason being is that lazy vals are not free, they require memory and synchronization.

Reply:
Sure, I can give it a try!